cygnus-common/src/main/java/com/telefonica/iot/cygnus/backends/postgresql/PostgreSQLCache.java

```diff
@@ -0,0 +1,101 @@
+ */
+
+package com.telefonica.iot.cygnus.backends.postgresql;
+
+import com.telefonica.iot.cygnus.log.CygnusLogger;
+import java.util.ArrayList;
+import java.util.HashMap;
+
+/**
+ *
+ * @author pcoello25
+ */
+public class PostgreSQLCache {
+    
+    private final CygnusLogger LOGGER = new CygnusLogger(PostgreSQLCache.class);
+    private HashMap<String, ArrayList<String>> cache = new HashMap<String, ArrayList<String>>();
+    
+    public HashMap<String, ArrayList<String>> getCache() {
+        return cache;
+    } // getCache
+    
+    public void setCache(HashMap<String, ArrayList<String>> PSQLcache) {
+        this.cache = PSQLcache;
+    } // setCache
+    
+    public boolean isSchemaInCache (String schemaName) {    
+        LOGGER.debug("Checking if the schema (" + schemaName + ") exists");
+        if (!cache.isEmpty()) {
+            
+            for (HashMap.Entry<String,ArrayList<String>> entry : cache.entrySet()) {
+                String schema = entry.getKey();
+                
+                if (schema.equals(schemaName)) {
+                    LOGGER.debug("Schema (" + schemaName + ") exists in Cache");
+                    return true;
+                } // if
+                
+            } // for
+            
+            LOGGER.debug("Schema (" + schemaName + ") doesnt' exist in Cache");
+            return false;
+        } else {
+            LOGGER.debug("Cache is empty");
+            return false;
+        } // if else
+
+    } // isSchemaInCache
+    
+    public boolean isTableInCachedSchema (String schemaName, String tableName) {
+        if (!cache.isEmpty()) {
+            
+            if (isSchemaInCache(schemaName)) {
+                ArrayList<String> tables = cache.get(schemaName);
+                
+                LOGGER.info("Checking if the table (" + tableName + ") belongs to (" + schemaName + ")");
+                
+                if (tables.contains(tableName)) {
+                    LOGGER.debug("Table (" + tableName + ") was found in the specified schema (" + schemaName + ")");
+                    return true;
+                } else {
+                    LOGGER.debug("Table (" + tableName + ") wasn't found in the specified schema (" + schemaName + ")");
+                    return false;
+                } // if else
+                
+            } // if             
+            
+            LOGGER.debug("Schema (" + schemaName + ") wasn't found in Cache");
+            return false;            
+        } else {
+            LOGGER.debug("Cache is empty");
+            return false;
+        }
+    } // isTableInCachedSchema
+    
+    public void persistSchemaInCache(String schemaName) {
+        cache.put(schemaName, new ArrayList<String>());
+    } // persistSchemaInCache
+    
+    public void persistTableInCache(String schemaName, String tableName) {
+        ArrayList<String> tableNames = cache.get(schemaName);
```

Comment:
Instead of doing this, which loops twice (one in order to check if the schema exists, another in order to put the table), try the following: iterate on the schemas, and once found add the table. If not found, nothing is done.